### DISCUSSION TITLE
Import from Nordpass CSV does not start when clicking import button

### DISCUSSION THREAD

**Discussion by loovanloon** [dani-garcia/vaultwarden#2843] [Q&A]
**Steps To Reproduce**

- Export your vault from Nordpass (version 4.32.8.31079) as a CSV-file to your local drive
- Open Vaultwarden in your browser
- Go to the import section of Vaultwarden
- Select the Nordpass profile
- Select the Nordpass-export CSV (see step 1)
- Click on 'import button'

**Expected Result**
The import function starts and the Nordpass export CSV file gets imported.

**Actual Result**

- The import does not start.
- The Nordpass-export CSV file is not imported
- The only visible feedback is the button-down-up-effect from the import button when you click it.

**Screenshots or Videos**
No response

**Additional Context**
When I imported the nordpass-export-csv-file with a non-nordpass profile an import-action took place, but resulted in the creation of folders in Bitwarden. I can’t remember which profile I took (one of the major players I guess)

I have also set up an official  free personal Bitwarden account. I tried to import Nordpass-csv with this official account and it gave the same results as I reported in this bug report. I also filed a bug report at Bitwarden  [https://github.com/bitwarden/clients/issues/3847](url)

**Operating System**
Windows

**Operating System Version**
Windows 11 Pro - 22H2 - build 22621.675

**Web Browser**
Chrome, Microsoft Edge, Firefox

**Browser Version**
Microsoft Edge Versie 106.0.1370.47 (Officiële build) (64-bits), 
Chrome is up-to-date Versie 106.0.5249.119 (Officiële build) (64-bits), 
Firefox 105.0.3 (64-bits), 
Firefox Developper 107.0b2 (64-bit)

**Build Version**
Bitwarden Inc. (Powered by Vaultwarden) Versie 2022.10.0

**Comment by BlackDex** [comment]
Thanks for the very detailed report 
As this is a client side issue, this is not something which is fixable by the Vaultwarden project.

Since you already reported this upstream at Bitwarden there is nothing left to do but wait until there is a fix.

**Comment by loovanloon** [comment]
In the browser console I found out that the problem is on my server configuration (Error: WebSocket failed to connect.) I’ll have to fix that.

PLEASE CLOSE BUG REPORT

**Reply by BlackDex** [reply]
I doubt it's anything to do with websockets, they are not in play during import/export. You can even disable websockets.

**Reply by loovanloon** [reply]
As requested I tested with nordpass Web CSV (version 4.32.10). Import did not work: same result.

**Console log output:**

```
consoleLog.service.ts:51 
        
       TypeError: Cannot read properties of null (reading 'length')
    at Ce.setCardExpiration (baseImporter.ts:327:24)
    at nordpassCsvImporter.ts:69:16
    at Array.forEach (<anonymous>)
    at Ce.parse (nordpassCsvImporter.ts:41:13)
    at ut.<anonymous> (import.service.ts:107:41)
    at Generator.next (<anonymous>)
    at 290.df3b8945ca6bd783424b.js:1:154065
    at new t (zone.js:1351:25)
    at ct (290.df3b8945ca6bd783424b.js:1:153810)
    at ut.import (290.df3b8945ca6bd783424b.js:1:154406)
```

**Reply by BlackDex** [reply]
I think you are mixing Vaultwarden and Bitwarden here 😉
This report should go to your closed ticket at Bitwarden.
